File: src/main/java/com/nanaios/polygonal_tech/registries/PolygonalTechBlockRegister.java
package com.nanaios.polygonal_tech.registries;

import com.nanaios.polygonal_tech.registration.impl.DeferredBlockRegister;
import com.nanaios.polygonal_tech.registration.impl.MultipleBlockRegister;
import net.minecraft.world.level.block.Block;
import net.minecraftforge.registries.RegistryObject;

public class PolygonalTechBlockRegister {
    public static MultipleBlockRegister BLOCKS = new MultipleBlockRegister();
    public static DeferredBlockRegister MACHINE_BLOCKS = BLOCKS.create().setBlockItemRegister(
            PolygonalTechItemRegister.BLOCK_ITEMS
    );

    public static RegistryObject<Block> PHOTOLYSIS_MACHINE_MK1;

    static {
            PHOTOLYSIS_MACHINE_MK1 = MACHINE_BLOCKS.registerGuiMachine(PolygonalTechNamedTokens.PHOTOLYSIS_MACHINE_MK1);
    }
}

File: src/main/kotlin/com/nanaios/polygonal_tech/main/registries/PolygonalTechBlockRegister.kt
package com.nanaios.polygonal_tech.main.registries

import com.nanaios.polygonal_tech.lib.block.TileBlock
import com.nanaios.polygonal_tech.lib.register.multi.DeferredBlockRegister
import com.nanaios.polygonal_tech.main.PolygonalTech
import net.minecraft.world.level.block.state.BlockBehaviour

object PolygonalTechBlockRegister {
    val MACHINE_BLOCKS = DeferredBlockRegister(PolygonalTech.MOD_ID)
    val TEST_MACHINE = MACHINE_BLOCKS.register("test_block") {location ->
        TileBlock(location, BlockBehaviour.Properties.of())
    }
}